aodntools/timeseries_products/velocity_aggregated_timeseries.py

```diff
@@ -0,0 +1,373 @@
+    obs_byte_template = {'datatype': np.byte, 'zlib': True, 'dimensions': ('OBSERVATION'), 'fill_value': 99}
+    obs_int_template = {'datatype': np.uint32, 'zlib': True, 'dimensions': ('OBSERVATION')}
+    inst_S256_template = {'datatype': 'str', 'dimensions': ('INSTRUMENT')}
+    inst_float_template ={'datatype': np.float32, 'dimensions': ('INSTRUMENT')}
+    inst_double_template ={'datatype': np.float64, 'dimensions': ('INSTRUMENT')}
+
+
+
+    UCUR = ds.createVariable(varname='UCUR', **obs_float_template)
+    VCUR = ds.createVariable(varname='VCUR', **obs_float_template)
+    WCUR = ds.createVariable(varname='WCUR', **obs_float_template)
+    DEPTH = ds.createVariable(varname='DEPTH', **obs_float_template)
+    UCURqc = ds.createVariable(varname='UCUR_quality_control', **obs_byte_template)
+    VCURqc = ds.createVariable(varname='VCUR_quality_control', **obs_byte_template)
+    WCURqc = ds.createVariable(varname='WCUR_quality_control', **obs_byte_template)
+    DEPTHqc = ds.createVariable(varname='DEPTH_quality_control', **obs_byte_template)
+    TIME = ds.createVariable(varname='TIME', **obs_double_template)
+    instrument_index = ds.createVariable(varname='instrument_index', **obs_int_template)
+
+    source_file = ds.createVariable(varname='source_file', **inst_S256_template)
+    instrument_id = ds.createVariable(varname='instrument_id', **inst_S256_template)
+    LATITUDE = ds.createVariable(varname='LATITUDE', **inst_double_template)
+    LONGITUDE = ds.createVariable(varname='LONGITUDE', **inst_double_template)
+    NOMINAL_DEPTH = ds.createVariable(varname='NOMINAL_DEPTH', **inst_float_template)
+    SECONDS_TO_MIDDLE = ds.createVariable(varname='SECONDS_TO_MIDDLE', **inst_float_template)
+
+    ## main loop
+    for index, file in enumerate(files_to_agg):
+        print(index)
+        with xr.open_dataset(os.path.join(input_dir, file)) as nc:
+
+            ## in water data only
+            nc = in_water(nc)
+
+            start = sum(varlen_list[:index + 1])
+            end = sum(varlen_list[:index + 2])
+            n_cells = get_number_flatvalues(nc)[1]
+            UCUR[start:end] = flat_variable(nc, 'UCUR')
+            UCURqc[start:end] = flat_variable(nc, 'UCUR_quality_control')
+            VCUR[start:end] = flat_variable(nc, 'VCUR')
+            VCURqc[start:end] = flat_variable(nc, 'VCUR_quality_control')
+            if 'WCUR' in nc.data_vars:
+                WCUR[start:end] = flat_variable(nc, 'WCUR')
+                WCURqc[start:end] = flat_variable(nc, 'WCUR_quality_control')
+            else:
+                WCUR[start:end] = np.full(varlen_list[index], np.nan)
+                WCURqc[start:end] = np.full(varlen_list[index], 9)
+            ##calculate depth
+            if 'HEIGHT_ABOVE_SENSOR' in nc.dims:
+                DEPTH[start:end] = (nc.DEPTH - nc.HEIGHT_ABOVE_SENSOR).values.flatten()
+                DEPTHqc[start:end] = np.array(n_cells * [nc.DEPTH_quality_control.values]).flatten()
```

Comment:
Looks a bit simpler and consistent with how you did the same operation for `TIME` and `instrument_index` below
```suggestion
                DEPTHqc[start:end] = np.repeat(nc.DEPTH_quality_control, n_cells).values
```

Reply:
yes indeed